types/src/transaction/signature_verified_transaction.rs

```diff
@@ -57,6 +57,14 @@ impl SignatureVerifiedTransaction {
             SignatureVerifiedTransaction::Invalid(_) => panic!("Expected valid transaction"),
         }
     }
+
+    pub fn valid_for_simulation(txn: SignedTransaction) -> Option<Self> {
+        // Simulation transactions should not carry valid signatures, otherwise
+        // malicious full nodes may execute them without user's explicit permission.
+        txn.verify_signature()
+            .err()
+            .map(|_| SignatureVerifiedTransaction::Valid(Transaction::UserTransaction(txn)))
+    }
```

Comment:
Would be good if we can approach simulation in the future without needing the public key or signature verification at all.  Basically a VM that lets you run arbitrary things, for test purposes only.